packages/x-bundle/DOCUMENTATION.md

```diff
@@ -970,6 +970,116 @@ export default {
 };
 ```
 
+### Caching
+
+#### Usage
+
+in order to improve your app's performance, you can use our caching:
+inside your resolver, use the cache Executor and feed it the chain of functions as an array,
+
+```ts
+{
+  Query: [
+    [],
+    {
+      UsersFind: [X.Cache([X.ToNova(UsersCollection)])],
+      UsersCount: [X.ToCollectionCount(UsersCollection)],
+    },
+  ];
+}
+```
+
+you can also use unique caching configuration for every resolver method by pasing object of options,
+the options fields are :
+| field | type | default | description |
+|--------------------------|----------|-------------|-----------------------------------------------------------------------------------------------------------------|
+| ttl | number | 30 | expiration time of cached Data in seconds |
+| refresh | boolean | false | if true :reset ttl count of a cached data, every time this data is consumed from the cache |
+| userBoundness | boolean | true | if true: takes into account other fields in order to personalize cache usage for every user/role.. |
+| userBoundnessFields | string[] | ["userId"] | if userBoundness is true: the fields we want to build on the user boundness. those fields expected to be in ctx |
+| expirationBoundness | boolean | true | if true: takes into account the expiration time of the data |
+| expirationBoundnessField | string | "expiredAt" | if expirationBoundness is true: the field of the expiration date or duration in seconds, Date or number |
+
+```ts
+{
+  Query: [
+    [],
+    {
+      UsersFind: [
+        X.Cache([X.ToNova(UsersCollection)], {
+          ttl: 30,
+          refresh: false,
+          userBoundness: true,
+          userBoundnessFields: ["userId"],
+          expirationBoundness: true,
+          expirationBoundnessField: "expiredAt",
+        }),
+      ],
+    },
+  ];
+}
+```
+
+#### Configuration
+
+the integrated default configuration of the caching is:
+
+```ts
+cacheConfig:{
+      store: "memory",
```

Comment:
proper indentation. please add vscode prettier autoformatting. use it like const cacheConfig = { } so it's valid ts